### ISSUE TITLE
osdsdock fails to retrieve pool list for ceph mimic

### ISSUE THREAD

**Issue by thatsdone** [opensds/opensds#989]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

osdsdock failed to retrieve list of pools from ceph while I was checking if opensds can work with ceph mimic (following slack discussion) using opensds-installer.
(I don't think this is an opensds-installer issue).

**What you expected to happen**:

osdsdock can get a list of pools from ceph, and 'osdsctl pool list' shows configured ceph pools.
In case of opensds-installer, the pool is 'rbd'.

**How to reproduce it (as minimally and precisely as possible)**:

Use opensds-installer. 

1. Modify 'ceph_stable_release:'  in 'ansible/group_vars/ceph/all.yml'  from luminous to mimic
2. Modify 'version:'  of 'download ceph-ansible source code' task in 'ansible/roles/osdsdock/scenarios/ceph.yml' from 'stable-3.0' to 'stable-3.1'
3. Modify other necessary variables appropriately according to your environment.

**Anything else we need to know?**:

```
* Ceph pool (created by opensds-installer) indeed exists.

[ubuntu@ubuntu201 opensds-installer(opensds_admin)]$ sudo ceph osd lspools
1 rbd

* osdsctl pool list shows empty.

[ubuntu@ubuntu201 opensds-installer(opensds_admin)]$ osdsctl pool list
+----+------+-------------+--------+---------------+--------------+
| Id | Name | Description | Status | TotalCapacity | FreeCapacity |
+----+------+-------------+--------+---------------+--------------+
+----+------+-------------+--------+---------------+--------------+

* osdsdock log shows pool list retrieval error.

[ubuntu@ubuntu201 opensds-installer(opensds_admin)]$ cat  /var/log/opensds/osdsdock.INFO
Log file created at: 2019/08/21 14:08:36
Running on machine: ubuntu201
Binary: Built with gc go1.11.2 for linux/amd64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0821 14:08:36.933016   21054 logs.go:75] [Info] LogFlushFrequency: 5s
I0821 14:08:36.935006   21054 dock.go:112] Dock server initialized! Start listening on port:[::]:50050
W0821 14:08:36.948688   21054 discovery.go:173] The pool of dock 91f94aae-94a7-5b97-9fd6-e07416e702f9 is empty!
E0821 14:08:36.948897   21054 dock.go:96] when calling capabilty report method:There is no pool can be found.


* ceph version/status/configuration

[ubuntu@ubuntu201 opensds-installer(opensds_admin)]$ ceph --version
ceph version 13.2.6 (7b695f835b03642f85998b2ae7b6dd093d9fbce4) mimic (stable)
[ubuntu@ubuntu201 opensds-installer(opensds_admin)]$ sudo ceph version
ceph version 13.2.6 (7b695f835b03642f85998b2ae7b6dd093d9fbce4) mimic (stable)

[ubuntu@ubuntu201 opensds-installer(opensds_admin)]$ sudo ceph -s
  cluster:
    id:     5b292b6f-5686-4bf2-853d-2ebe6fae68f7
    health: HEALTH_OK

  services:
    mon: 1 daemons, quorum ubuntu201
    mgr: ubuntu201(active)
    osd: 1 osds: 1 up, 1 in

  data:
    pools:   1 pools, 100 pgs
    objects: 0  objects, 0 B
    usage:   108 MiB used, 25 GiB / 25 GiB avail
    pgs:     100 active+clean

[ubuntu@ubuntu201 opensds-installer(opensds_admin)]$ cat /etc/ceph/ceph.conf
# Please do not change this file directly since it is managed by Ansible and will be overwritten

[global]
rbd default features = 1
fsid = 5b292b6f-5686-4bf2-853d-2ebe6fae68f7



mon initial members = ubuntu201
mon host = 192.168.1.201

public network = 192.168.1.201/24
cluster network = 192.168.1.201/24


[osd]
osd mkfs type = xfs
osd mkfs options xfs = -f -i size=2048
osd mount options xfs = noatime,largeio,inode64,swalloc
osd journal size = 5120
```

**Environment**:
- Hotpot(release/branch) version: Capri (v0.6.1)
- OS (e.g. from /etc/os-release): Ubuntu 16.04.6
- Kernel (e.g. `uname -a`): Linux ubuntu201 4.4.0-157-generic #185-Ubuntu SMP Tue Jul 23 09:17:01 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: openstack-installer (ansible, master branch)
- Others:

**Comment by wisererik** [comment]
Thanks, Itoh-san, @PravinRanjan10 @xxwjj please take a look at this issue~

**Comment by thatsdone** [comment]
I got a successful result using my opensds-installer branch below.

https://github.com/thatsdone/opensds-installer/tree/refactor-ceph-ansible

The essential point was to wait for ceph stabilization in 2 senses.
1.  ceph -s returns HEALTH_OK and all the defined osds become up.
2. all the expected ceph pools become visible using 'rados df' which is equivalent to a low level API of go-ceph and osdsdock uses.

The branch above implements step1 and step2 of https://github.com/opensds/opensds-installer/issues/270 including the above 2 points.

I'm preparing a PR.

**Comment by thatsdone** [comment]
I'd be happy if anyone can test the branch. Reproduce procedure is the following.

1. Tested distro is Ubuntu 16.04.06 LTS (amd64)
   * requires python-requests and jq.
2. Please use https://github.com/thatsdone/opensds-installer/tree/refactor-ceph-ansible
    * Please note the above is still under development (mostly cleanup works for PR).
3. Ensure you have one additional volume (in my case /dev/vdb) and make it clean.
3. Modifications I made are the following. 
   * The below assumes all-in-one configuration including ceph.
```
[ubuntu@ubuntu203 ansible(opensds_admin)]$ git diff
diff --git a/ansible/group_vars/ceph/all.yml b/ansible/group_vars/ceph/all.yml
index c0cc427..a7988d4 100644
--- a/ansible/group_vars/ceph/all.yml
+++ b/ansible/group_vars/ceph/all.yml
@@ -25,11 +25,12 @@ dummy:

 ceph_origin: repository
 ceph_repository: community
-ceph_stable_release: luminous
+ceph_stable_release: mimic
 public_network: "{{ ansible_default_ipv4.address }}/24"
 cluster_network: "{{ public_network }}"
 monitor_interface: "{{ ansible_default_ipv4.interface }}"
 devices:
+  - '/dev/vdb'
   #- '/dev/sda'
   #- '/dev/sdb'
 osd_scenario: collocated
diff --git a/ansible/group_vars/common.yml b/ansible/group_vars/common.yml
index 70daa6f..92f5bed 100644
--- a/ansible/group_vars/common.yml
+++ b/ansible/group_vars/common.yml
@@ -23,11 +23,11 @@ dummy:
 ###########

 # This field indicates local machine host ip
-host_ip: 127.0.0.1
+host_ip: 192.168.1.203

 # This field indicates which project should be deploy
 # 'hotpot', 'gelato' or 'all'
-deploy_project: all
+deploy_project: hotpot

 # This field indicates which way user prefers to install, currently support
 # 'repository', 'release' and 'container'
diff --git a/ansible/group_vars/osdsdb.yml b/ansible/group_vars/osdsdb.yml
index ae5625b..ea04d0a 100644
--- a/ansible/group_vars/osdsdb.yml
+++ b/ansible/group_vars/osdsdb.yml
@@ -46,7 +46,7 @@ etcd_dir: /opt/etcd-{{ etcd_release }}-linux-amd64
 # DOCKER  #
 ###########

-etcd_container_enabled: false
+etcd_container_enabled: true

 # These fields are NOT suggested to be modified
 etcd_docker_image: quay.io/coreos/etcd:latest
diff --git a/ansible/group_vars/osdsdock.yml b/ansible/group_vars/osdsdock.yml
index 4a58b95..a3df933 100644
--- a/ansible/group_vars/osdsdock.yml
+++ b/ansible/group_vars/osdsdock.yml
@@ -25,7 +25,8 @@ dummy:
 # Change it according to your backend, currently support 'lvm', 'ceph', 'cinder', 'nfs'
 # DISABLE OR COMMENT "enabled_backends: lvm" IF YOU WANT TO INSTALL DIFFERENT BACKENDS ON MULTI-NODES AND mention it in "local.hosts" file
 # Comment this part if you want to use different backends on different nodes
-enabled_backends: lvm,nfs #For Multi-backend add backends here, for eg. enabled_backends: lvm,ceph,cinder,nfs
+#enabled_backends: lvm,nfs #For Multi-backend add backends here, for eg. enabled_backends: lvm,ceph,cinder,nfs
+enabled_backends: ceph

 # Change it according to your node type (host or target), currently support
 # 'provisioner', 'attacher'
@@ -65,7 +66,7 @@ ceph_description: This is a ceph backend service
 ceph_driver_name: ceph
 ceph_config_path: "{{ opensds_driver_config_dir }}/ceph.yaml"

-ceph_ansible_branch: stable-3.0
+ceph_ansible_branch: stable-3.1

 ceph_check_interval: 30
 ceph_check_count: 12
```

**Comment by wisererik** [comment]
it's solved with PR #273, will close it.
